### PR TITLE
Reorganize and clean up version information and init()

### PR DESCRIPTION
**Pull request by jacobmcnamee**
Also adds v3 HDL version check

### CODE REVIEW

File: src/board/v3/nap/nap_common.c [hunk 1]
```diff
@@ -26,8 +26,7 @@
 
 #define PROCESS_PERIOD_ms (1000)
 
-void nap_isr(void *context);
-void nap_unlock(const u8 key[]);
```
Comment (wltr) [inline src/board/v3/nap/nap_common.c]: You might need this forward declaration if you want to unlock the NAP in `nap_setup()`

File: src/board/v3/init.c [hunk 2]
```diff
@@ -55,38 +63,68 @@ static void reset_callback_register(void)
...
 
 /* Check NAP authentication status. Block and print error message
  * if authentication has failed. This must be done after the NAP,
  * USARTs, and SBP subsystems are set up, so that SBP messages and
  * be sent and received (it can't go in init() or nap_setup()).
  */
-void check_nap_auth(void)
+static void nap_auth_check(void)
 {
+
```
Comment (wltr) [inline src/board/v3/init.c]: In the latest NAP version, `NAP->STATUS` bit 0 is 1 as long as the NAP is locked.

**Comment by wltr** [comment]
lgtm

**Comment by gsmcmullin** [comment]
lgtm

**Pull request merged**